torch/_dynamo/variables/builder.py

```diff
@@ -2807,7 +2821,7 @@ def update_dim2constraint(dim, constraint_range, name):
 
         # Reflect the user directive in the frame_state
         # For dynamic, apply None always
-        if marked_dynamic:
+        if marked_dynamic or source.name() in dynamic_sources:
```

Comment:
the log bellow 
log.debug("automatic dynamic %s marked dynamic", name) would be confusing .mmm thinking